scheduler_error_mailer/models/ir_cron.py

```diff
@@ -25,28 +25,26 @@ class IrCron(models.Model):
         ),
     )
 
-    @api.model
-    def _handle_callback_exception(
-        self, cron_name, server_action_id, job_id, job_exception
-    ):
-        res = super()._handle_callback_exception(
-            cron_name, server_action_id, job_id, job_exception
-        )
-        my_cron = self.browse(job_id)
-
-        if my_cron.email_template_id:
+    def _handle_callback_exception(self, cron_name, server_action_id, job_exception):
+        self.ensure_one()
+        if self.email_template_id:
             # we put the job_exception in context to be able to print it inside
             # the email template
             context = {"job_exception": str(job_exception), "dbname": self._cr.dbname}
 
             _logger.debug("Sending scheduler error email with context=%s", context)
 
-            template = my_cron.email_template_id.with_context(**context).sudo()
-            template.send_mail(my_cron.id, force_send=True)
-
-        return res
+            template = self.email_template_id.with_context(**context).sudo()
+            template.send_mail(self.id, force_send=True)
 
     @api.model
     def _test_scheduler_failure(self):
         """This function is used to test and debug this module."""
         raise UserError(_("Task failure with UID = %d.") % self._uid)
+
+    @api.model
+    def _callback(self, cron_name, server_action_id):
+        try:
+            return super()._callback(cron_name, server_action_id)
+        except Exception as e:
+            self._handle_callback_exception(cron_name, server_action_id, e)
```

Comment:
Is it intended that the exception is not raised again?

maybe we also should move the _handle_callback_exception in this method when odoo also removed it?

Reply:
I think this is correct as it is. The Odoo base _callback() only logs any exception, not raising.
And I honestly can't see any reason for moving _handle_callback_exception() inside here.

Reply:
I checked again and indeed the Odoo base _callback() is raising the exception.
I added this commit to re-raise it again:
https://github.com/OCA/server-tools/pull/3084/commits/31459b46d67fbce434394940690ded476b9f07a2